File: 0415-Add-Strings/0415.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        index1 = len(num1) - 1
        index2 = len(num2) - 1
        ans = ""

        carry = 0
        while index1 >= 0 or index2 >= 0:
            if index1 >= 0 and index2 >= 0:
                add = int(num1[index1]) + int(num2[index2]) + carry
            elif index1 >= 0:
                add = int(num1[index1]) + carry
            elif index2 >= 0:
                add = int(num2[index2]) + carry

            if add >= 10:
                carry = 1
                add %= 10
                ans = str(add) + ans
            else:
                ans = str(add) + ans
                carry = 0

            index1 -= 1
            index2 -= 1
        
        if carry != 0:
            ans = "1" + ans

        return ans

        

File: 0415-Add-Strings/0415.java
class Solution {
    public String addStrings(String num1, String num2) {
        int index1 = num1.length() - 1, index2 = num2.length() - 1;
        int carry = 0;
        String ans = "";

        while (index1 >= 0 || index2 >= 0) {
            int x = 0, y = 0;
            if (index1 >= 0) {
                x = num1.charAt(index1) - 48;
                index1--;
            }
            if (index2 >= 0) {
                y = num2.charAt(index2) - 48;
                index2--;
            }

            int add = x + y + carry;
            System.out.println(add);
            if (add >= 10) {
                carry = 1;
                ans = String.valueOf(add % 10) + ans;
            } else {
                ans = String.valueOf(add) + ans;
                carry = 0;
            }
        }

        if (carry == 1) {
            ans = "1" + ans;
        }

        return ans;

    }
}